indexing-solr/src/main/java/edu/unc/lib/boxc/indexing/solr/filter/SetDatastreamFilter.java

```diff
@@ -129,29 +132,120 @@ private String getExtent(List<BinaryObject> binList) {
 
             if (fitsMd != null) {
                 Element imgMd = fitsMd.getChild("image", FITS_NS);
-
                 if (imgMd != null) {
                     String imgHeight = imgMd.getChildTextTrim("imageHeight", FITS_NS);
                     String imgWidth = imgMd.getChildTextTrim("imageWidth", FITS_NS);
-                    if (!StringUtils.isBlank(imgHeight) && !StringUtils.isBlank(imgWidth)) {
-                        try {
-                            Integer.parseInt(imgHeight);
-                            Integer.parseInt(imgWidth);
-                            extent = imgHeight + "x" + imgWidth;
-                        } catch (NumberFormatException e) {
-                            log.warn("Invalid image width or height from FITS {}: {} x {}",
-                                    fits.getPid().getQualifiedId(), imgWidth, imgHeight);
-                        }
-                    }
+                    return formatDimensionExtent(imgHeight, imgWidth, fits.getPid().getQualifiedId());
+                }
+
+                Element videoMd = fitsMd.getChild(FITS_VIDEO_NAME, FITS_NS);
+                if (videoMd != null) {
+                    return formatVideoExtent(videoMd, fits.getPid().getQualifiedId());
+                }
+
+                Element audioMd = fitsMd.getChild("audio", FITS_NS);
+                if (audioMd != null) {
+                    var audioTime = formatTime(audioMd);
+                    return "xx" + audioTime;
                 }
             }
-            return extent;
+
+            return null;
         } catch (RepositoryException | FedoraException e) {
             log.warn("Unable to parse FITS for {}", fitsId, e);
             return null;
+        } catch (NumberFormatException e) {
+            log.warn("Unable to parse audio/video track time for {}", fitsId, e);
+            return null;
         }
     }
 
+    private String formatVideoExtent(Element videoMd, String pid) {
+        var videoInfo = videoMd;
+        var trackInfo = videoMd.getChildren("track", FITS_NS);
+
+        var numTracks = trackInfo.size();
+        if (numTracks > 0) {
+            var videoTrack = 0;
+            for (int i = 0; i < numTracks; i++) {
+                var type = trackInfo.get(i).getAttributeValue("type");
+                if (type.equals(FITS_VIDEO_NAME)) {
```

Comment:
```suggestion
                if (FITS_VIDEO_NAME.equals(type)) {
```
Just to be safe in case FITS decides to be inconsistent, since getAttributeValue will return null if the attribute isn't present. Or we can use `Object.equals(FITS_VIDEO_NAME, type)`